File: src/api/helpers/week-number.ts
import { getUTCTime } from "~/api/private/dates";

export const translateToPronoteWeekNumber = (dateToTranslate: Date, startDay: Date): number => {
  const daysDiff = Math.floor((getUTCTime(dateToTranslate) - getUTCTime(startDay)) / (1000 * 60 * 60 * 24));
  return 1 + Math.floor(daysDiff / 7);
};

File: library/src/commonMain/kotlin/api/helpers/week-number.kt
package api.helpers

import kotlinx.datetime.LocalDateTime
import kotlinx.datetime.TimeZone
import kotlinx.datetime.toInstant
import kotlin.math.floor

fun translateToWeekNumber (dateToTranslate: LocalDateTime, startDay: LocalDateTime): Int {
    val daysDiff = (dateToTranslate.toInstant(TimeZone.UTC) - startDay.toInstant(TimeZone.UTC)).inWholeDays
    return 1 + floor(daysDiff.toDouble() / 7).toInt()
}